### PR TITLE
Redirect trait

### PR DESCRIPTION
**Pull request by tristanward**
This trait overrides the `resolveRouteBinding()` method and checks for a redirect in the Spatie `MissingPageRedirector` package if the model is not found, before returning a 404.

### CODE REVIEW

**Review by m2de — CHANGES_REQUESTED** [review]
Looks great, might seem negligible but I am not sure the trait name reflects what is does very well? I try to think of the name if I was to use it in a sentence. E.g. `My Author Model Can Be Redirected` ... which doesn't tell me a lot. Maybe something like `My Author Model Should Redirect Missing Pages` for example would make it easier to understand when reading the code later on? Thoughts?

**Comment by crumb1e** [comment]
@m2de I agree! Naming things in plain English clears up any confusion when it comes to implementing these traits.

**Comment by tristanward** [comment]
Thanks @m2de and @crumb1e completely see where you are coming from.  Updates pushed!

**Review by m2de — APPROVED** [review]
